cmd/controller/ctrl.go

```diff
@@ -190,17 +216,28 @@ func getExistingResource(uuid string, listDR *apis.DiskList) *apis.Disk {
 }
 
 // addNewDiskResource add the newly identified disks to the etcd
-func addNewDiskResource(c *Controller, detected v1.BlockDeviceInfo, listDR *apis.DiskList) {
-	for _, blk := range detected.Blockdevices {
-		if blk.Type == "disk" {
-			uuid := getUid(blk)
-			n, err := strconv.ParseInt(blk.Size, 10, 64)
+func addNewDiskResource(c *Controller, devices []*udev.Udevice, listDR *apis.DiskList) {
+	for i := range devices {
+		device := devices[i]
+		if device.Devtype() == "disk" && device.PropertyValue("SCSI_TYPE") == "disk" {
```

Comment:
`disk` can be used as a constant string
also all the keys which are used to get disk info